### DISCUSSION TITLE
Не тестируются и не извлекаются файлы из запароленых архивов 7z

### DISCUSSION THREAD

**Discussion by michel-nk** [elfmz/far2l#2146] [General]
Система: Kubuntu 22.04. Far2l установлен из репозитория на launchpad (последнее обновление - 2-3 дня назад). Создаю архив 7z с паролем. Создается успешно. Получается зайти внутрь и прочитать каталог (содержимое). Но при попытке извлечь любой файл получаю сообщение (жирным выделил имя извлекаемого файла):

_7z Decoder 22.01 (x64) : Igor Pavlov : Public domain : 2022-07-15
Extracting **settings/AltShiftV_macro_dev.ini**
ERROR: decoder doesn't support this archive_

Аналогично при попытке тестирования архива по Alt+F3 -> "Тестирование". При этом системные программы-архиваторы нормально тестируют этот архив и извлекают содержимое т.е. архив с паролем создан нормально.

С форматом zip такой проблемы нет - и архивируется и тестируется и извлекается нормально.
Как победить проблему?

P.S. Только недавно узнал, что существует far2l. Восторгам нет предела!!! Разработчики, благодарю вас от всей души. Под виндой Far - мой основной инструмент для работы с файлами на протяжении более 20-ти лет. Именно так - я влюбился в программу ещё начиная с её первых версий. И тут вдруг такое открытие! Ура!!!
Когда больше половины моих компов переехало под крыло Линкуса, именно Far'а мне не хватало больше всего. Долго мучал/приучал себя к DoubleCommander'у, худо-бедно за несколько месяцев привык ... и тут бац!!! Вот он!!!! Самый настоящий Far!!!! Ура-ура-ура! Тряхнул стариной, вспомнил макроязык и за пару дней восстановил все мои любимые макросы. Теперь опять всё летает безо всяких мышей как в старые добрые времена. Короче - песня.

Есть ещё несколько вопросов и предложений к разработчикам, но это уже потом. Пока только хочу выразить благодарность и озвучить самый обидный косяк с 7z.

**Comment by akruphi** [comment]
Посмотрите в F9->Options->Plugins configuration->Archive support->Command lines->7Z
Если в начале команды стоит **^**, то используется вкомпилированный модуль архиватора. Без - внешний доступный из системы, что немного медленнее. Возможно просто убрать **^** даст эффект. Иначе поиграйтесь с параметрами командной строки. К сожалению как видно из `man 7z` часть параметров у 7z не интуитивны или не полностью делают желаемое - про это многократно ворчали  в #1901 , #1882, #1370 и т.п.

**Reply by michel-nk** [reply]
Спасибо! Действительно помогло - убрал ^ и всё заработало. Имеющийся в системе 7z 16.02 теперь спрашивает пароль.

Ещё поэкспериментировал. Не только убрал ^, но и в строки настройки архиватора вместо просто 7z (системный вызов), вписал полный путь к версии 23.01 из домашней папки. Тоже всё отлично работает - запароленые архивы и создаются и при извлечении нормально запрашивается пароль и всё работает.
Так что подозреваю дело не столько в ключах, сколько во "встроенной версии" 7z.

Хочу ещё подчеркнуть, что никакие другие параметры в командных строках не менял. Исключительно путь к ...
